tools/ontology-builder/src/compute_descendent_mappings.py

```diff
@@ -0,0 +1,279 @@
+ created to simplify referencing in the Single Cell Data Portal.
+
+The descendant mappings should be updated when:
+
+1. The ontology version is updated,
+2. A new tissue or cell type is added to the production corpus, or,
+3. The hand-curated systems, organs, cell classes or cell subclasses are updated.
+"""
+
+import json
+import os
+from typing import Any, Dict, List
+from urllib.request import urlopen
+
+import env
+from cellxgene_ontology_guide.ontology_parser import OntologyParser
+
+
+def load_prod_datasets() -> Any:
+    """
+    Request datasets the production corpus.
+    """
+    response = urlopen("https://api.cellxgene.cziscience.com/dp/v1/datasets/index").read().decode("utf-8")
+    return json.loads(response)
+
+
+def extract_cell_types(datasets: List[Dict[str, Any]]) -> List[str]:
+    """
+    List the set of cell type values for the given datasets.
+
+    :param datasets: a list of datasets from the production corpus.
+    :return: a list formated of cell type values
+    """
+    cell_types = set()
+    for dataset in datasets:
+        for cell_type in dataset["cell_type"]:
+            cell_types.add(cell_type["ontology_term_id"].replace("_", ":", False))
+    return list(cell_types)
+
+
+def extract_tissues(datasets: List[Dict[str, Any]]) -> List[str]:
+    """
+    List the set of tissue values for the given datasets.
+
+    :param datasets: a list of datasets from the production corpus.
+    :return: a list of formated tissue values with tags for tissue type.
+    """
+    tissues = set()
+    for dataset in datasets:
+        for tissue in dataset["tissue"]:
+            formatted_entity_name = tissue["ontology_term_id"].replace("_", ":", False)
+            tissue_type = tissue.get("tissue_type")
+            tissues.add(tag_tissue_type(formatted_entity_name, tissue_type))
+
+    return list(tissues)
+
+
+def tag_tissue_type(entity_name: str, tissue_type: str) -> str:
+    """
+    Append the tissue type to the given entity name if the tissue type is cell
+    culture or organoid, otherwise return the entity name as is.
+
+    :param entity_name: str entity name
+    :param tissue_type: str tissue type
+    :return: str entity name with tissue type appended
+    """
+    # Tissue types
+    tissue_type_cell_culture = "cell culture"
+    tissue_type_organoid = "organoid"
+
+    # Handle error case (possible if tissue has not been migrated to 4.0.0+ schema).
+    if tissue_type is None:
+        return entity_name
+
+    if tissue_type == tissue_type_cell_culture:
+        # true if the given tissue type is "cell culture".
+        return f"{entity_name} ({tissue_type_cell_culture})"
+
+    if tissue_type == tissue_type_organoid:
+        # true if the given tissue type is "organoid".
+        return f"{entity_name} ({tissue_type_organoid})"
+
+    return entity_name
+
+
+def key_organoids_by_ontology_term_id(entity_names: List[str]) -> Dict[str, str]:
```

Comment:
is there an equivalent need for cell culture? if not, why do we also tag cell culture terms?